### DISCUSSION TITLE
Can't build Whatsapp 2.23.18.79 on mac with apktool 2.81

### DISCUSSION THREAD

**Discussion by ScubaInstructor** [iBotPeaches/Apktool#3332] [Q&A] [answered]
I am able to decompile whatsapp

% ./apktool.sh d com.whatsapp_2.23.18.79.apk
I: Using Apktool 2.8.1 on com.whatsapp_2.23.18.79.apk
I: Loading resource table...
I: Decoding AndroidManifest.xml with resources...
I: Loading resource table from file: /Users/user-home/Library/apktool/framework/1.apk
I: Regular manifest package...
I: Decoding file-resources...
I: Decoding values */* XMLs...
I: Baksmaling classes.dex...
I: Baksmaling classes2.dex...
I: Baksmaling classes3.dex...
I: Baksmaling classes4.dex...
I: Baksmaling classes5.dex...
I: Copying assets and libs...
I: Copying unknown files...
I: Copying original files...
I: Copying META-INF/services directory

Trying to build it:% ./apktool.sh b com.whatsapp_2.23.18.79.apk
I: Using Apktool 2.8.1
brut.directory.DirectoryException: file must be a directory: com.whatsapp_2.23.18.79.apk
(base) user-home@machine Development % ./apktool.sh b com.whatsapp_2.23.18.79    
I: Using Apktool 2.8.1
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes3 folder into classes3.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes4 folder into classes4.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes5 folder into classes5.dex...
I: Checking whether sources has changed...
I: Smaling smali_classes2 folder into classes2.dex...
I: Checking whether resources has changed...
I: Building resources...
W: /Users/user-home/Development/com.whatsapp_2.23.18.79/res/layout-v22/APKTOOL_DUMMYVAL_0x7f0d0008.xml:2: error: Error parsing XML: not well-formed (invalid token)
W: 
W: /Users/user-home/Development/com.whatsapp_2.23.18.79/res/layout-v26/APKTOOL_DUMMYVAL_0x7f0d0017.xml:5: error: Error parsing XML: not well-formed (invalid token)
W: 
W: /Users/user-home/Development/com.whatsapp_2.23.18.79/res/layout-v22/APKTOOL_DUMMYVAL_0x7f0d0022.xml:4: error: Error parsing XML: not well-formed (invalid token)

............... (many more)

W: /Users/user-home/Development/com.whatsapp_2.23.18.79/res/color/APKTOOL_DUMMYVAL_0x7f0608a7.xml:6: error: No resource identifier found for attribute 'state_dragged' in package 'com.whatsapp'
W: 
W: /Users/user-home/Development/com.whatsapp_2.23.18.79/res/color/APKTOOL_DUMMYVAL_0x7f0608a8.xml:5: error: No resource identifier found for attribute 'state_dragged' in package 'com.whatsapp'
W: 
brut.androlib.exceptions.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/var/folders/0c/cz2ppcns1fg99s69_psng4v80000gn/T/brut_util_Jar_177762655669710072979045226120467006228.tmp, p, --forced-package-id, 127, --min-sdk-version, 21, --target-sdk-version, 33, --version-code, 231879002, --version-name, 2.23.18.79, --no-version-vectors, -F, /var/folders/0c/cz2ppcns1fg99s69_psng4v80000gn/T/APKTOOL8350739175835821353.tmp, -e, /var/folders/0c/cz2ppcns1fg99s69_psng4v80000gn/T/APKTOOL17066476374896597120.tmp, -0, arsc, -I, /Users/user-home/Library/apktool/framework/1.apk, -S, /Users/user-home/Development/com.whatsapp_2.23.18.79/res, -M, /Users/user-home/Development/com.whatsapp_2.23.18.79/AndroidManifest.xml]

Any suggestions?
Am I missing any tools on my machine?

**Comment by iBotPeaches** [comment, marked as answer]
Yep apologies known issue at the moment here - https://github.com/iBotPeaches/Apktool/issues/3129
